backend/src/main/java/gov/cdc/usds/simplereport/idp/authentication/LiveOktaAuthentication.java

```diff
@@ -0,0 +1,225 @@
+   *     <p>https://developer.okta.com/docs/reference/api/authn/#request-example-for-activation-token
+   */
+
+  /**
+    * {
+  "stateToken": "005Oj4_rx1yAYP2MFNobMXlM2wJ3QEyzgifBd_T6Go",
+  "expiresAt": "2017-03-29T21:35:47.000Z",
+  "status": "PASSWORD_RESET",
+  "recoveryType": "ACCOUNT_ACTIVATION",
+  "_embedded": {
+    "user": {
+      "id": "00ub0oNGTSWTBKOLGLNR",
+      "passwordChanged": "2015-09-08T20:14:45.000Z",
+      "profile": {
+        "login": "dade.murphy@example.com",
+        "firstName": "Dade",
+        "lastName": "Murphy",
+        "locale": "en_US",
+        "timeZone": "America/Los_Angeles"
+      }
+    },
+    "policy": {
+      "expiration": {
+        "passwordExpireDays": 5
+      },
+      "complexity": {
+        "minLength": 8,
+        "minLowerCase": 1,
+        "minUpperCase": 1,
+        "minNumber": 1,
+        "minSymbol": 0
+      }
+    }
+  },
+  "_links": {
+    "next": {
+      "name": "resetPassword",
+      "href": "https://${yourOktaDomain}/api/v1/authn/credentials/reset_password",
+      "hints": {
+        "allow": [
+          "POST"
+        ]
+      }
+    },
+    "cancel": {
+      "href": "https://${yourOktaDomain}/api/v1/authn/cancel",
+      "hints": {
+        "allow": [
+          "POST"
+        ]
+      }
+    }
+  }
+}
+
+HTTP/1.1 401 Unauthorized
+Content-Type: application/json
+
+{
+  "errorCode": "E0000004",
+  "errorSummary": "Authentication failed",
+  "errorLink": "E0000004",
+  "errorId": "oae2fYYJmkuTg-NyozqBkb3sw",
+  "errorCauses": []
+}
+
+
+curl -v -X POST \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-H "Authorization: SSWS ${api_token}" \
+-H "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36" \
+-H "X-Forwarded-For: 23.235.46.133" \
+-d '{
+  "token": "o7AFoTGE9xjQiHQK6dAa"
+}' "https://${yourOktaDomain}/api/v1/authn"
+
+    */
+  public String activateUser(String activationToken, String crossForwardedHeader, String userAgent)
+      throws InvalidActivationLinkException {
+    System.out.println("activating user (printed)");
+    LOG.info("activating user");
+    JSONObject requestBody = new JSONObject();
+    requestBody.put("token", activationToken);
+    String authorizationToken = "SSWS " + _apiToken;
+    RestTemplate restTemplate =
+        new RestTemplateBuilder(
+                rt ->
+                    rt.getInterceptors()
+                        .add(
+                            (request, body, execution) -> {
+                              HttpHeaders headers = request.getHeaders();
+                              headers.setContentType(MediaType.APPLICATION_JSON);
+                              headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+                              headers.add(HttpHeaders.USER_AGENT, userAgent);
+                              headers.add("X-Forwarded-For", crossForwardedHeader);
```

Comment:
Does Okta reject requests if the IP and user agent don't look like authentic browser traffic? Making a forwarded IP chain and a user agent part of the method signature is a strong signal that we should only be calling the method in response to a user-initiated action (i.e., never from a script), which may or may not be what we want. If Okta is placing such a requirement on the API action, then it makes sense to take this approach, but if not, this approach may be limiting in the long run.

Reply:
From what I can tell, Okta rejects all traffic to this endpoint unless you've specifically allowlisted the final forwarding IP address (in our case, the app's IP). I can't imagine a scenario where we'd want to call this from a script, so I think this should continue to be safe to use.

This is the documentation chain I'm looking at:
The actual API call definition: https://developer.okta.com/docs/reference/api/authn/#primary-authentication-with-activation-token
The IP API requirements: https://developer.okta.com/docs/reference/api-overview/#ip-address
How Okta handles IP chains: https://help.okta.com/en/prod/Content/Topics/Security/network/about-ip-zones.htm